### ISSUE TITLE
fix(testing): test_hydra_sweep_ddp_sim — limit_val_batches=0.1 * DDP processes yields 0 batches (same as #46)

### ISSUE THREAD

**Issue by ktinubu** [tinaudio/synth-setter#626]
## Problem

`tests/test_sweeps.py::test_hydra_sweep_ddp_sim` fails end-to-end under DDP sim on GPU CI because `+trainer.limit_val_batches=0.1` combined with the DDP sharding of a small validation set produces 0 batches on each rank. Lightning rejects this as a `MisconfigurationException` before `trainer.fit` can make progress.

## Root cause

Lightning shards the val loader across DDP ranks before applying the `limit_val_batches` fraction. With a small val loader (5 batches) and a fraction of `0.1`, the per-rank slice rounds to 0 batches:

```
lightning.fabric.utilities.exceptions.MisconfigurationException:
  You requested to check 0.1 of the `val_dataloader` but 0.1 * 5 < 1.
  Please increase the `limit_val_batches` argument. Try at least `limit_val_batches=0.2`.
```

The sweep test sets all three fraction-based limits at `tests/test_sweeps.py:70-72`:

```python
"+trainer.limit_train_batches=0.01",
"+trainer.limit_val_batches=0.1",
"+trainer.limit_test_batches=0.1",
```

## Sibling fix

This is the same class of bug as #46, which was fixed for `test_train_ddp_sim` in #515 by switching the fraction limits to integer limits (e.g. `limit_val_batches=1`). The fix template was never carried over to the sweep test in `tests/test_sweeps.py`.

Relevant snippet from PR #515 (applied to `test_train_ddp_sim`):

```python
# Integer limits avoid the `fraction * batches < 1` error when the
# fixture's fractions (0.01, 0.1) shrink too far under DDP sharding.
cfg_train.trainer.limit_train_batches = 1
cfg_train.trainer.limit_val_batches = 1
cfg_train.trainer.limit_test_batches = 1
```

## Evidence

Level 1 verification of #621 (which fixed the `trainer.max_epochs` struct-mode error) showed that the sweep now launches and DDP-sim initializes both processes, but training then fails on the `limit_val_batches` issue. See the Level 1 verification comment on PR #621: https://github.com/tinaudio/synth-setter/pull/621#issuecomment-4281310366

This issue is a sibling to #620 (also tracking a test_hydra_sweep_ddp_sim failure surfaced by #621's verification).

## Acceptance criteria

- [ ] `tests/test_sweeps.py::test_hydra_sweep_ddp_sim` passes end-to-end on GPU CI after switching the three fraction limits to integer limits.
- [ ] The non-DDP `test_hydra_sweep` is left untouched (its fractions work fine without DDP sharding).
- [ ] The skipped tests (`test_optuna_sweep`, `test_optuna_sweep_ddp_sim_wandb`) are left untouched — #514 owns those rewrites.
